pkg/hintrunner/hint.go

```diff
@@ -87,3 +88,79 @@ func (hint TestLessThan) Execute(vm *VM.VirtualMachine) error {
 
 	return nil
 }
+
+type WideMul128 struct {
+	lhs  ResOperander
+	rhs  ResOperander
+	high CellRefer
+	low  CellRefer
+}
+
+func (hint WideMul128) String() string {
+	return "WideMul128"
+}
+
+func (hint WideMul128) Execute(vm *VM.VirtualMachine) error {
+	u128Max := big.NewInt(1).Lsh(big.NewInt(1), 128)
+	u128Max = big.NewInt(1).Sub(u128Max, big.NewInt(1))
+	u128MaxFelt := &f.Element{}
+	u128MaxFelt.SetBigInt(u128Max)
+
+	lsh, err := hint.lhs.Resolve(vm)
+	if err != nil {
+		return fmt.Errorf("resolve lhs operand %s: %v", hint.lhs, err)
+	}
+
+	rhs, err := hint.rhs.Resolve(vm)
+	if err != nil {
+		return fmt.Errorf("resolve rhs operand %s: %v", hint.rhs, err)
+	}
+
+	lhsFelt, err := lsh.ToFieldElement()
+	if err != nil {
+		return err
+	}
+	if lhsFelt.Cmp(u128MaxFelt) > 0 {
+		return fmt.Errorf("lhs operand %s should be u128", lhsFelt)
+	}
+
+	rhsFelt, err := rhs.ToFieldElement()
+	if err != nil {
+		return err
+	}
+	if rhsFelt.Cmp(u128MaxFelt) > 0 {
+		return fmt.Errorf("lhs operand %s should be u128", rhsFelt)
+	}
+
+	mul := big.NewInt(1).Mul(lhsFelt.BigInt(big.NewInt(1)), rhsFelt.BigInt(big.NewInt(1)))
+
+	low := big.NewInt(1)
+	high := big.NewInt(1)
+	low.And(mul, u128Max)
+	high.Rsh(mul, 128)
+
+	lowFelt := &f.Element{}
+	lowFelt.SetBigInt(low)
+	highFelt := &f.Element{}
+	highFelt.SetBigInt(high)
```

Comment:
I realized now that we need a library for handling multiplication for values in the u256 range. And make that type easily convertible into Field Element as well. It is ok as it is right now, just some thoughts.

Reply:
Indeed, we might be able to optimize all this, I implemented it in quite a simplistic way